src/performance/metrics/StatisticCollector.cpp

```diff
@@ -46,64 +48,172 @@ int StatisticCollector::getMemoryUsageByProcess() {
         }
     }
     fclose(file);
-    std::cout << "Memory Usage: " + std::to_string(result) << std::endl;
+    // std::cout << "Memory Usage: " + std::to_string(result) << std::endl;
     return result;
 }
 
-int StatisticCollector::parseLine(char* line) {
-    int i = strlen(line);
-    const char* p = line;
-    while (*p < '0' || *p > '9') p++;
-    line[i - 3] = '\0';
-    i = atoi(p);
-    return i;
+int StatisticCollector::getThreadCount() {
+    FILE *file = fopen("/proc/self/stat", "r");
+    long result;
+    char line[128];
```

Comment:
Can we move 128 to a constant? I see several places where 128 is used below.

Reply:
Replaced all with constant